config.go

```diff
@@ -44,18 +64,68 @@ var globalConfig = &Config{
 	},
 }
 
-func (c *Config) LoadConfig() error {
+func LoadConfig(configFile string) (*Config, error) {
+	// Load config file as YAML if provided
+	if configFile != "" {
+		buf, err := os.ReadFile(configFile)
+		if err != nil {
+			return nil, fmt.Errorf("error reading config file: %s", err)
+		}
+		err = yaml.Unmarshal(buf, globalConfig)
+		if err != nil {
+			return nil, fmt.Errorf("error parsing config file: %s", err)
+		}
+	}
 	// Load config values from environment variables
 	// We use "dummy" as the app name here to (mostly) prevent picking up env
 	// vars that we hadn't explicitly specified in annotations above
-	err := envconfig.Process("dummy", c)
+	err := envconfig.Process("dummy", globalConfig)
 	if err != nil {
-		return fmt.Errorf("error processing environment: %s", err)
+		return nil, fmt.Errorf("error processing environment: %s", err)
 	}
-	return nil
+	// Populate NetworkMagic from named networks
+	if err := globalConfig.populateNetworkMagic(); err != nil {
+		return nil, err
+	}
+	return globalConfig, nil
 }
 
 // GetConfig returns the global config instance
 func GetConfig() *Config {
 	return globalConfig
 }
+
+func (c *Config) populateNetworkMagic() error {
+	if c.Node.NetworkMagic != uint32(0) {
+		return nil
+	}
+	if c.App.Network != "" {
+		switch c.App.Network {
+		case "preview":
+			c.Node.NetworkMagic = libada.Preview.ProtocolMagic()
+		case "preprod":
+			c.Node.NetworkMagic = libada.Preprod.ProtocolMagic()
+		case "testnet":
+			c.Node.NetworkMagic = libada.Testnet.ProtocolMagic()
+		case "mainnet":
+			c.Node.NetworkMagic = libada.Mainnet.ProtocolMagic()
+		default:
+			return fmt.Errorf("unknown network: %s", c.App.Network)
+		}
+	}
+	if c.Node.Network != "" {
+		switch c.Node.Network {
+		case "preview":
+			c.Node.NetworkMagic = libada.Preview.ProtocolMagic()
+		case "preprod":
+			c.Node.NetworkMagic = libada.Preprod.ProtocolMagic()
+		case "testnet":
+			c.Node.NetworkMagic = libada.Testnet.ProtocolMagic()
+		case "mainnet":
+			c.Node.NetworkMagic = libada.Mainnet.ProtocolMagic()
+		default:
+			return fmt.Errorf("unknown network: %s", c.Node.Network)
+		}
+	}
+	return nil
+}
```

Comment:
FYI, there's a helper function in gOuroboros to look up a well-known network by name, and then you can retrieve the network magic value from the returned data

Reply:
Cool. I lifted this from `tx-submit-api` directly, so I'll update both.